src/sentry/integrations/github/webhook.py

```diff
@@ -136,30 +142,26 @@ def _handle(self, event, organization, repo):
                         except CommitAuthor.DoesNotExist:
                             commit_author = None
 
-                        if commit_author is not None and not is_anonymous_email(
+                        if commit_author is not None and not self.is_anonymous_email(
                             commit_author.email
                         ):
                             author_email = commit_author.email
                             gh_username_cache[gh_username] = author_email
                         else:
                             try:
-                                gh_user = client.get('/users/%s' % gh_username)
+                                gh_user = client.get_user(gh_username)
                             except ApiError as exc:
                                 logger.exception(six.text_type(exc))
                             else:
                                 # even if we can't find a user, set to none so we
                                 # don't re-query
                                 gh_username_cache[gh_username] = None
                                 try:
-                                    user = User.objects.filter(
-                                        social_auth__provider='github',
-                                        social_auth__uid=gh_user['id'],
-                                        org_memberships=organization,
-                                    )[0]
-                                except IndexError:
+                                    identity = Identity.objects.get(external_id=gh_user['id'])
```

Comment:
hmm there should definitely be an additional filter here since `external_id` isn't unique across providers. i think you could add `idp__type=self.provider` or something similar and that should be good at least for github? GHE you may need to actually filter on the `idp` not just the type cc @EvanPurkhiser

Reply:
added in `idp__type=self.provider` for now but not sure the best way to add in for GHE and would probably need to make a `get_identity` method or something?